hphp/runtime/vm/jit/unique-stubs.cpp

```diff
@@ -1118,12 +1123,26 @@ TCA emitEnterTCHelper(CodeBlock& cb, DataBlock& data, UniqueStubs& us) {
 #endif
 
   return vwrap2(cb, cb, data, [&] (Vout& v, Vout& vc) {
-    // Native func prologue.
-    v << stublogue{true};
+    bool saveFrame = true;
 
     // Architecture-specific setup for entering the TC.
     v << inittc{};
 
+    // Set if FP need to be saved.
+    switch (arch()) {
+      case Arch::ARM:
+      case Arch::X64:
+        break;
+      case Arch::PPC64:
+        // Do no save the FP for PPC64, as no new frame is created in the
+        // stublogue
+        saveFrame = false;
+        break;
+    }
+
+    // Native func prologue.
+    v << stublogue{saveFrame};
```

Comment:
The point of have the `stublogue{true}` in this case is that there may have information pushed on the stack, so a new frame need to be created, in order to set its end. I think the same happens for `x64`, that is why the parameter is `true` for this case.
On PPC64 This is different when compared with the VM entering, because entering the VM the frame was already created and set, here the frame was allocated, but rvmfp was not set, that is why the parameter need to be `true` here.